File: julia8024/PGS_ 12981.py
def solution(n, words):
    answer = []
    
    for idx, val in enumerate(words):
        if val in words[:idx] or (idx != 0 and words[idx-1][-1] != val[0]):
            answer.append(idx % n + 1)
            answer.append(idx // n + 1)
            break

    return [0, 0] if len(answer) == 0 else answer

File: chaeryeon823/PGS_12981.js
// 영어 끝말잇기
// 문자열

function solution(n, words) {
  var answer = [0, 0];
  let last = words[0].slice(-1);
  let set = new Set();
  set.add(words[0]);
  for (let i = 1; i < words.length; i++) {
    let arr = words.slice(0, i);
    if (last !== words[i][0] || set.has(words[i])) {
      answer[0] = (i % n) + 1;
      answer[1] = Math.floor(i / n) + 1;
      break;
    }
    last = words[i].slice(-1);
    set.add(words[i]);
  }
  return answer;
}
